### PR TITLE
Update pre-commit hooks guide

### PR DESCRIPTION
**Pull request by bdice**
## Description
Updates contributing guide to remove outdated information about clang-format, which is now a pre-commit hook.

Context: I recently modified some other RAPIDS repos to use pre-commit as part of the migration to GitHub Actions, and I recognized we could improve these docs.

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

**Review by robertmaynard — APPROVED** [review]

**Comment by vyasr** [comment]
Will take a look at this PR in the new year! @bdice feel free to ping me if I don't get to it in early Jan.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -377,9 +363,29 @@ Now code linters and formatters will be run each time you commit changes.
...
 
+### Summary of pre-commit hooks
+
+The following section describes some of the core pre-commit hooks used by the repository.
+See `.pre-commit-config.yaml` for a full list.
+
+C++/CUDA is formatted with [`clang-format`](https://clang.llvm.org/docs/ClangFormat.html).
+
+Additionally, many editors have plugins or extensions that you can set up to automatically run
+`clang-format` either manually or on file save.
```
Comment (vyasr) [inline CONTRIBUTING.md:374]: I would move this sentence out of the pre-commit section. It's useful to know and belongs in the contributing guide, but not here. Maybe in a separate second-level header `# Format on save` or something? I think you can implement something similar with black too, or at least I recall some other devs mentioning it (no need to spend time figuring it out if you don't know offhand though).

**Review by vyasr — APPROVED** [review]
Never mind just realized how small this was and reviewed it.

**Pull request merged**